docs/How-to-Revendor-a-Dependency.md

```diff
@@ -0,0 +1,46 @@
+# How to Revendor a Dependency
+
+At the time of writing, Homebrew vendors several dependencies within the
+[Library/Homebrew/vendor](https://github.com/Homebrew/brew/tree/master/Library/Homebrew/vendor)
+subdirectory.
+
+This page serves to document the process for revendoring these dependencies.
+
+## Copying the files
+
+Each directory in `vendor/` contains the **contents** of the `lib/` directory for the corresponding
+dependency. Thus, `require`ing a vendored dependency looks like this:
+
+```ruby
+require "vendor/library/file"
+```
+
+Thus, the copying process is as simple as removing the old files, and copying the new files into
+place.
+
+Here's an example for the imaginary "foobar" vendored dependency:
+
+```bash
```

Comment:
I’d suggest to prepend this script with the following line:

```bash
git rm -rf --cached /path/to/brew/Library/Homebrew/vendor/foobar
```

This will clean the Git index from obsolete files, i. e. files which are not present in the updated copy.
(Even though it’s an extra step, it makes `git add`ing the new copy much less confusing.)